File: PrevailingTrend‌/数据库配置/database_config.py
"""
数据库配置文件
统一管理所有模块的数据库配置
"""

import os
from typing import Dict, Any
from dotenv import load_dotenv

# 加载环境变量
load_dotenv()

class DatabaseConfig:
    """数据库配置类"""
    
    # 基础配置
    HOST = os.getenv('DB_HOST', 'localhost')
    PORT = int(os.getenv('DB_PORT', 3306))
    USERNAME = os.getenv('DB_USERNAME', 'root')
    PASSWORD = os.getenv('DB_PASSWORD', 'rr1234RR')
    DATABASE = os.getenv('DB_DATABASE', 'pt')
    CHARSET = os.getenv('DB_CHARSET', 'utf8mb4')
    
    # 连接池配置
    MAX_CONNECTIONS = int(os.getenv('DB_MAX_CONNECTIONS', 20))
    MIN_CONNECTIONS = int(os.getenv('DB_MIN_CONNECTIONS', 5))
    CONNECTION_TIMEOUT = int(os.getenv('DB_CONNECTION_TIMEOUT', 30))
    
    @classmethod
    def get_connection_config(cls) -> Dict[str, Any]:
        """获取数据库连接配置"""
        return {
            'host': cls.HOST,
            'port': cls.PORT,
            'user': cls.USERNAME,
            'password': cls.PASSWORD,
            'database': cls.DATABASE,
            'charset': cls.CHARSET,
            'autocommit': True
        }
    
    @classmethod
    def get_connection_string(cls) -> str:
        """获取数据库连接字符串"""
        return f"mysql+pymysql://{cls.USERNAME}:{cls.PASSWORD}@{cls.HOST}:{cls.PORT}/{cls.DATABASE}?charset={cls.CHARSET}"

# 表名配置 - 按层级和模块组织
class TableNames:
    """表名配置类"""
    
    # 第一层模块 - 数据采集层
    LAYER1 = {
        # 万得行业分类
        'wind_industry': {
            'industry_classification': 'l1_wind_industry_classification',
            'stock_industry_mapping': 'l1_wind_stock_industry_mapping'
        },
        # 公司名字列表
        'company_list': {
            'company_info': 'l1_company_list_info',
            'market_classification': 'l1_company_market_classification'
        },
        # 国内热点数据
        'domestic_hotspot': {
            'hotspot_data': 'l1_domestic_hotspot_data',
            'hotspot_analysis': 'l1_domestic_hotspot_analysis'
        },
        # 国外热点数据
        'foreign_hotspot': {
            'hotspot_data': 'l1_foreign_hotspot_data',
            'hotspot_analysis': 'l1_foreign_hotspot_analysis'
        },
        # 腾讯济安指数
        'tencent_index': {
            'index_data': 'l1_tencent_index_data',
            'index_history': 'l1_tencent_index_history'
        },
        # 雪球等论坛热点数据
        'forum_hotspot': {
            'forum_data': 'l1_forum_hotspot_data',
            'topic_analysis': 'l1_forum_topic_analysis'
        },
        # 其他互联网信息
        'internet_info': {
            'info_data': 'l1_internet_info_data',
            'info_analysis': 'l1_internet_info_analysis'
        }
    }
    
    # 第二层模块 - 数据处理层
    LAYER2 = {
        'company_attributes': {
            'company_attributes': 'l2_company_attributes',
            'attribute_analysis': 'l2_company_attribute_analysis'
        },
        'hotspot_data': {
            'hotspot_summary': 'l2_hotspot_data_summary',
            'hotspot_trends': 'l2_hotspot_trends'
        }
    }
    
    # 第三层模块 - 数据整合层
    LAYER3 = {
        'qcc_data': {
            'company_qcc_data': 'l3_qcc_company_data',
            'qcc_analysis': 'l3_qcc_analysis'
        },
        'tax_bank_report': {
            'tax_data': 'l3_tax_bank_report_data',
            'tax_analysis': 'l3_tax_analysis'
        },
        'financial_statements': {
            'balance_sheet': 'l3_financial_balance_sheet',
            'income_statement': 'l3_financial_income_statement',
            'cash_flow': 'l3_financial_cash_flow'
        },
        'forum_data': {
            'forum_summary': 'l3_forum_data_summary',
            'sentiment_analysis': 'l3_forum_sentiment_analysis'
        }
    }
    
    # 第四层模块 - 数据评分层
    LAYER4 = {
        'company_score': {
            'company_score': 'l4_company_score',
            'score_history': 'l4_company_score_history'
        },
        'industry_score': {
            'industry_score': 'l4_industry_score',
            'score_history': 'l4_industry_score_history'
        },
        'industry_company_score': {
            'industry_company_score': 'l4_industry_company_score',
            'score_history': 'l4_industry_company_score_history'
        }
    }
    
    # 第五层模块 - 权重配置层
    LAYER5 = {
        'factor_weights': {
            'factor_weights': 'l5_factor_weights',
            'weight_history': 'l5_factor_weight_history'
        }
    }
    
    # 第六层模块 - 预测分析层
    LAYER6 = {
        'curve_prediction': {
            'prediction_data': 'l6_curve_prediction_data',
            'prediction_analysis': 'l6_curve_prediction_analysis'
        }
    }
    
    # 系统表
    SYSTEM = {
        'system_logs': 'system_logs',
        'data_flow_logs': 'system_data_flow_logs',
        'module_status': 'system_module_status',
        'data_quality': 'system_data_quality'
    }
    
    @classmethod
    def get_table_name(cls, layer: str, module: str, table: str) -> str:
        """获取表名"""
        layer_tables = getattr(cls, layer.upper(), {})
        module_tables = layer_tables.get(module, {})
        return module_tables.get(table, f"{layer}_{module}_{table}")
    
    @classmethod
    def get_all_tables(cls) -> Dict[str, Dict[str, str]]:
        """获取所有表名"""
        return {
            'LAYER1': cls.LAYER1,
            'LAYER2': cls.LAYER2,
            'LAYER3': cls.LAYER3,
            'LAYER4': cls.LAYER4,
            'LAYER5': cls.LAYER5,
            'LAYER6': cls.LAYER6,
            'SYSTEM': cls.SYSTEM
        }

# 创建全局配置实例
db_config = DatabaseConfig()
table_names = TableNames() 

File: PrevailingTrend‌/数据库配置/DatabaseConfig.java
package com.prevailingtrend.config;

import java.io.IOException;
import java.io.InputStream;
import java.util.HashMap;
import java.util.Map;
import java.util.Properties;

/**
 * 数据库配置类
 * 统一管理所有模块的数据库配置
 * 
 * @author 大势所趋风险框架团队
 * @version 1.0
 */
public class DatabaseConfig {
    
    private static DatabaseConfig instance;
    private Properties properties;
    
    // 基础配置常量
    public static final String DEFAULT_HOST = "localhost";
    public static final int DEFAULT_PORT = 3306;
    public static final String DEFAULT_USERNAME = "root";
    public static final String DEFAULT_PASSWORD = "rr1234RR";
    public static final String DEFAULT_DATABASE = "pt";
    public static final String DEFAULT_CHARSET = "utf8mb4";
    
    // 连接池配置常量
    public static final int DEFAULT_MAX_CONNECTIONS = 20;
    public static final int DEFAULT_MIN_CONNECTIONS = 5;
    public static final int DEFAULT_CONNECTION_TIMEOUT = 30;
    
    private DatabaseConfig() {
        loadProperties();
    }
    
    /**
     * 获取单例实例
     */
    public static DatabaseConfig getInstance() {
        if (instance == null) {
            synchronized (DatabaseConfig.class) {
                if (instance == null) {
                    instance = new DatabaseConfig();
                }
            }
        }
        return instance;
    }
    
    /**
     * 加载配置文件
     */
    private void loadProperties() {
        properties = new Properties();
        
        // 尝试加载配置文件
        try (InputStream is = getClass().getClassLoader().getResourceAsStream("database.properties")) {
            if (is != null) {
                properties.load(is);
            }
        } catch (IOException e) {
            System.err.println("警告：无法加载database.properties文件，使用默认配置");
        }
    }
    
    /**
     * 获取数据库主机地址
     */
    public String getHost() {
        return properties.getProperty("db.host", DEFAULT_HOST);
    }
    
    /**
     * 获取数据库端口
     */
    public int getPort() {
        return Integer.parseInt(properties.getProperty("db.port", String.valueOf(DEFAULT_PORT)));
    }
    
    /**
     * 获取数据库用户名
     */
    public String getUsername() {
        return properties.getProperty("db.username", DEFAULT_USERNAME);
    }
    
    /**
     * 获取数据库密码
     */
    public String getPassword() {
        return properties.getProperty("db.password", DEFAULT_PASSWORD);
    }
    
    /**
     * 获取数据库名称
     */
    public String getDatabase() {
        return properties.getProperty("db.database", DEFAULT_DATABASE);
    }
    
    /**
     * 获取字符集
     */
    public String getCharset() {
        return properties.getProperty("db.charset", DEFAULT_CHARSET);
    }
    
    /**
     * 获取最大连接数
     */
    public int getMaxConnections() {
        return Integer.parseInt(properties.getProperty("db.max.connections", String.valueOf(DEFAULT_MAX_CONNECTIONS)));
    }
    
    /**
     * 获取最小连接数
     */
    public int getMinConnections() {
        return Integer.parseInt(properties.getProperty("db.min.connections", String.valueOf(DEFAULT_MIN_CONNECTIONS)));
    }
    
    /**
     * 获取连接超时时间
     */
    public int getConnectionTimeout() {
        return Integer.parseInt(properties.getProperty("db.connection.timeout", String.valueOf(DEFAULT_CONNECTION_TIMEOUT)));
    }
    
    /**
     * 获取JDBC连接URL
     */
    public String getJdbcUrl() {
        return String.format("jdbc:mysql://%s:%d/%s?characterEncoding=%s&useSSL=false&allowMultiQueries=true&serverTimezone=Asia/Shanghai",
                getHost(), getPort(), getDatabase(), getCharset());
    }
    
    /**
     * 获取连接配置映射
     */
    public Map<String, Object> getConnectionConfig() {
        Map<String, Object> config = new HashMap<>();
        config.put("host", getHost());
        config.put("port", getPort());
        config.put("username", getUsername());
        config.put("password", getPassword());
        config.put("database", getDatabase());
        config.put("charset", getCharset());
        config.put("jdbcUrl", getJdbcUrl());
        config.put("maxConnections", getMaxConnections());
        config.put("minConnections", getMinConnections());
        config.put("connectionTimeout", getConnectionTimeout());
        return config;
    }
}

/**
 * 表名配置类
 * 按层级和模块组织表名
 */
class TableNames {
    
    // 第一层模块 - 数据采集层
    public static class Layer1 {
        // 万得行业分类
        public static final String WIND_INDUSTRY_CLASSIFICATION = "l1_wind_industry_classification";
        public static final String WIND_STOCK_INDUSTRY_MAPPING = "l1_wind_stock_industry_mapping";
        
        // 公司名字列表
        public static final String COMPANY_LIST_INFO = "l1_company_list_info";
        public static final String COMPANY_MARKET_CLASSIFICATION = "l1_company_market_classification";
        
        // 国内热点数据
        public static final String DOMESTIC_HOTSPOT_DATA = "l1_domestic_hotspot_data";
        public static final String DOMESTIC_HOTSPOT_ANALYSIS = "l1_domestic_hotspot_analysis";
        
        // 国外热点数据
        public static final String FOREIGN_HOTSPOT_DATA = "l1_foreign_hotspot_data";
        public static final String FOREIGN_HOTSPOT_ANALYSIS = "l1_foreign_hotspot_analysis";
        
        // 腾讯济安指数
        public static final String TENCENT_INDEX_DATA = "l1_tencent_index_data";
        public static final String TENCENT_INDEX_HISTORY = "l1_tencent_index_history";
        
        // 雪球等论坛热点数据
        public static final String FORUM_HOTSPOT_DATA = "l1_forum_hotspot_data";
        public static final String FORUM_TOPIC_ANALYSIS = "l1_forum_topic_analysis";
        
        // 其他互联网信息
        public static final String INTERNET_INFO_DATA = "l1_internet_info_data";
        public static final String INTERNET_INFO_ANALYSIS = "l1_internet_info_analysis";
    }
    
    // 第二层模块 - 数据处理层
    public static class Layer2 {
        public static final String COMPANY_ATTRIBUTES = "l2_company_attributes";
        public static final String COMPANY_ATTRIBUTE_ANALYSIS = "l2_company_attribute_analysis";
        public static final String HOTSPOT_DATA_SUMMARY = "l2_hotspot_data_summary";
        public static final String HOTSPOT_TRENDS = "l2_hotspot_trends";
    }
    
    // 第三层模块 - 数据整合层
    public static class Layer3 {
        public static final String QCC_COMPANY_DATA = "l3_qcc_company_data";
        public static final String QCC_ANALYSIS = "l3_qcc_analysis";
        public static final String TAX_BANK_REPORT_DATA = "l3_tax_bank_report_data";
        public static final String TAX_ANALYSIS = "l3_tax_analysis";
        public static final String FINANCIAL_BALANCE_SHEET = "l3_financial_balance_sheet";
        public static final String FINANCIAL_INCOME_STATEMENT = "l3_financial_income_statement";
        public static final String FINANCIAL_CASH_FLOW = "l3_financial_cash_flow";
        public static final String FORUM_DATA_SUMMARY = "l3_forum_data_summary";
        public static final String FORUM_SENTIMENT_ANALYSIS = "l3_forum_sentiment_analysis";
    }
    
    // 第四层模块 - 数据评分层
    public static class Layer4 {
        public static final String COMPANY_SCORE = "l4_company_score";
        public static final String COMPANY_SCORE_HISTORY = "l4_company_score_history";
        public static final String INDUSTRY_SCORE = "l4_industry_score";
        public static final String INDUSTRY_SCORE_HISTORY = "l4_industry_score_history";
        public static final String INDUSTRY_COMPANY_SCORE = "l4_industry_company_score";
        public static final String INDUSTRY_COMPANY_SCORE_HISTORY = "l4_industry_company_score_history";
    }
    
    // 第五层模块 - 权重配置层
    public static class Layer5 {
        public static final String FACTOR_WEIGHTS = "l5_factor_weights";
        public static final String FACTOR_WEIGHT_HISTORY = "l5_factor_weight_history";
    }
    
    // 第六层模块 - 预测分析层
    public static class Layer6 {
        public static final String CURVE_PREDICTION_DATA = "l6_curve_prediction_data";
        public static final String CURVE_PREDICTION_ANALYSIS = "l6_curve_prediction_analysis";
    }
    
    // 系统表
    public static class System {
        public static final String SYSTEM_LOGS = "system_logs";
        public static final String DATA_FLOW_LOGS = "system_data_flow_logs";
        public static final String MODULE_STATUS = "system_module_status";
        public static final String DATA_QUALITY = "system_data_quality";
    }
    
    /**
     * 获取表名
     * @param layer 层级
     * @param module 模块
     * @param table 表名
     * @return 完整表名
     */
    public static String getTableName(String layer, String module, String table) {
        return String.format("%s_%s_%s", layer.toLowerCase(), module.toLowerCase(), table.toLowerCase());
    }
    
    /**
     * 获取所有表名的映射
     */
    public static Map<String, Map<String, String>> getAllTables() {
        Map<String, Map<String, String>> allTables = new HashMap<>();
        
        // 这里可以通过反射获取所有静态字段，但为简化起见，手动添加主要表
        Map<String, String> layer1Tables = new HashMap<>();
        layer1Tables.put("wind_industry_classification", Layer1.WIND_INDUSTRY_CLASSIFICATION);
        layer1Tables.put("company_list_info", Layer1.COMPANY_LIST_INFO);
        layer1Tables.put("domestic_hotspot_data", Layer1.DOMESTIC_HOTSPOT_DATA);
        
        Map<String, String> systemTables = new HashMap<>();
        systemTables.put("system_logs", System.SYSTEM_LOGS);
        systemTables.put("module_status", System.MODULE_STATUS);
        
        allTables.put("LAYER1", layer1Tables);
        allTables.put("SYSTEM", systemTables);
        
        return allTables;
    }
}